dbms/src/DataStreams/GraphiteRollupSortedBlockInputStream.cpp

```diff
@@ -225,8 +228,15 @@ void GraphiteRollupSortedBlockInputStream::startNextRow(ColumnPlainPtrs & merged
         merged_columns[j]->insertFrom(*cursor->all_columns[j], cursor->pos);
     }
 
-    if (!current_pattern)
+    if (next_pattern)
+    {
+        next_pattern->function->create(place_for_aggregate_state.data());
+        aggregate_state_created = true;
+    }
+    else
         merged_columns[value_column_num]->insertFrom(*cursor->all_columns[value_column_num], cursor->pos);
```

Comment:
Still incorrect - in case next_pattern is null, we must choose a value with the max version column, not the value at the current cursor.